src/style-spec/expression/definitions/index.js

```diff
@@ -172,7 +172,7 @@ CompoundExpression.register(expressions, {
     'feature-state': [
         ValueType,
         [StringType],
-        (ctx, [key]) => get(key.evaluate(ctx), ctx.state())
+        (ctx, [key]) => get(key.evaluate(ctx), ctx.featureState || {})
```

Comment:
Should there be a check for `feature` when using `featureState`? Feature state is only valid if there is a current feature. As it stands, it's possible to call `Expression#evaluate` with a `null` feature, but non-null `featureState.`